esp32_jpeg_enc/main/jpeg_decoder.c

```diff
@@ -0,0 +1,72 @@
+#include <stdio.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+
+#include <esp_log.h>
+#include <esp_err.h>
+
+#include "esp_audio_mem.h"
+#include "smtp_client.h"
+
+static const char *TAG="jpeg_decoding";
+
+
+extern const uint8_t raw_image_start[] asm("_binary_esp_img_start");
+extern const uint8_t raw_image_end[]   asm("_binary_esp_img_end");
+
+
+/*extern const uint8_t raw_image_start[] asm("_binary_raw_image_start");
+extern const uint8_t raw_image_end[]   asm("_binary_raw_image_end");*/
+
+int decode_jpeg(unsigned char *source_buf,
+                     unsigned long *source_buf_sz,
+                     unsigned char *out_buf
+                     );
+
+unsigned long in_buf_sz = 96 * 96 * 4;
+unsigned char in_buf[96 * 96 * 4];
+unsigned char out_buf[96 * 96 * 12];
+
+static void jpeg_decoder_task(void *pvParameters)
+{
+    (void) pvParameters;
+
+    /* Fill in image data. Random numbers for now. */
+    //for (int i = 0; i < image_width * image_height; i++) {
+    //    input_buffer[i] = raw_image_start[i];
+    //}
+
+    ESP_LOGI(TAG, "Decoding an image");
+
+    decode_jpeg((unsigned char *) raw_image_start,in_buf,in_buf_sz,out_buf);
+
+
+    /* sending an email */
+    send_email(out_buf);
+    vTaskDelete(NULL);
+}
+
```

Comment:
Instead of adding files to existing esp32_jpeg_enc which is for encoder, create new example: esp32_jpeg_dec.